src/shell/shell_cmds.cpp

```diff
@@ -149,7 +149,11 @@ bool DOS_Shell::execute_shell_cmd(char *name, char *arguments) {
 	SHELL_Cmd shell_cmd = {};
 	if (!lookup_shell_cmd(name, shell_cmd))
 		return false; // name isn't a shell command!
-	(this->*(shell_cmd.handler))(arguments);
+
+	// Hack to allow '-h' and '--help' in shell commands
+	std::regex re("(^| )(-h|--help)( |$)");
+	std::string tmp = std::regex_replace(arguments, re, "$1/?$3", std::regex_constants::format_first_only);
+	(this->*(shell_cmd.handler))(tmp.data());
```

Comment:
What it would take do do this in a non-hacky way?

Reply:
A previous iteration of this had a more proper way of doing this, but it requires changing `ScanCMDBool` (which only accepts `/` flags), because of *course* DOSBox has at least three different sets of functions for parsing command line options.

Reply:
> because of course DOSBox has at least three different sets of functions for parsing command line options.

Oof :face_with_spiral_eyes: 
On a positive note - more low hanging fruit to for sweet cleanups like this!

Reply:
Yeah, off the top of my head, I can think of the parsing of main DOSBox args on startup, Parsing options for internal shell commands (`ScanCMDBool` and friends), and the functions for parsing `Program` arguments (`FindExist` and friends).

Reply:
Nice round up - I didn't realize all three did some duplicate amount of argument parsing!

Reply:
Alright, I ended up doing the non-hacky approach.

In the process, the `HELP()` macro is gone :)